curriculum/challenges/english/25-front-end-development/lab-lunch-picker-program/66db529d37ad966480ebb633.md

```diff
@@ -0,0 +1,157 @@
+4. Once items have been added to the `adultsMeal`, you should use the appropriate array method to remove the first item. 
+5.  You should use the appropriate `Math` methods to generate a random number between `0` to the length of the `kidsMeal` array. Assign that result to a variable called `randomIndexKids`.
+6. You should use the appropriate `Math` methods to generate a random number between `0` to the length of the `kidsMeal` array. Assign that result to a variable called `randomIndexAdults`.
+7.  Log the following messages to the console:
+
+- `"Today kids should eat:"` followed by the selected meal to the console.
+- `"Today adults should eat:"` followed by the selected meal to the console.
+
+# --hints--
+
+You should declare a `kidsMeal` array.
+
+```js
+assert.isArray(kidsMeal);
+```
+
+You should declare an `adultsMeal` array.
+
+```js
+assert.isArray(adultsMeal);
+```
+
+Your `kidsMeal` array should contain atleast three items.
+
+```js
+const matched = __helpers.removeJSComments(code).match(/(const|let)\s+kidsMeal\s*=\s*\[(?<items>.+?)\]/)
+const kidsMealArray = matched.groups.items.split(",");
+assert.isAtLeast(kidsMealArray.length, 3);
+```
+
+Your `kidsMeal` array should contain only non-empty string values.
+
+```js
+const matched = __helpers.removeJSComments(code).match(/(const|let)\s+adultsMeal\s*=\s*\[(?<items>.+?)\]/)
+const adultsMealArray = matched.groups.items.split(",");
+assert.isAtLeast(adultsMealArray.length, 1); 
+
+for (let i = 0; i < kidsMeal.length; i++) {
+  assert.isString(kidsMeal[i]); // Check that every item is a string
+  assert.isAbove(kidsMeal[i].length, 0, 'String should not be empty'); // Check that string is not empty
+}
+```
+
+Your `adultsMeal` array should contain atleast three items.
+
+```js
+const matched = __helpers.removeJSComments(code).match(/(const|let)\s+adultsMeal\s*=\s*\[(?<items>.+?)\]/)
+const adultsMealArray = matched.groups.items.split(",");
+assert.isAtLeast(adultsMealArray.length, 3);
+```
+
+Your `adultsMeal` array should contain only non-empty string values.
+
+```js
+const matched = __helpers.removeJSComments(code).match(/(const|let)\s+adultsMeal\s*=\s*\[(?<items>.+?)\]/)
+const adultsMealArray = matched.groups.items.split(",");
+assert.isAtLeast(adultsMealArray.length, 1);
+
+for (let i = 0; i < adultsMeal.length; i++) {
+  assert.isString(adultsMeal[i]); 
+  assert.isAbove(adultsMeal[i].length, 0, 'String should not be empty'); 
+}
+```
+
+You should remove the last item from the `kidsMeal` array using `pop()`.
+
+```js
+assert.match(__helpers.removeJSComments(code), /kidsMeal\.pop/);
+```
+
+You should remove the first item from the `adultsMeal` array using `shift()`.
+
+```js
+assert.match(__helpers.removeJSComments(code), /adultsMeal\.shift/);
+```
+
+You should use `Math.random()` to generate a random number from `0` upto the length of the `kidsMeal` array that will be used to select a random meal from the `kidsMeal` array and store it in a variable called `randomIndexKids`.
+
```

Comment:
This test doesn't work properly. If I have `const randomIndexKids = Math.floor(Math.random() * 10)`, and when the test runs `randomIndexKids` is `1`, the  test does not fail. If you run it again it could be `9` and fail.

Also, it matches `Math.random()` without any relation with the assignment to `randomIndexKids`.

The same is true for the test after this one.

Reply:
I have updated both, let me know what you think @Dario-DC.